lib/cubit/ln_address/ln_address_cubit.dart

```diff
@@ -219,30 +270,193 @@ class LnAddressCubit extends Cubit<LnAddressState> {
+    } catch (e) {
+      _logger.severe('Failed to recover LNURL Webhook.', e);
+      rethrow;
+    }
+  }
+
+  /// Prepares register request & registers a webhook for a given public key.
+  Future<RegisterRecoverLnurlPayResponse> _prepareAndRegisterLnurlWebhook({
+    required String pubKey,
+    required String webhookUrl,
+    required String? baseUsername,
   }) async {
-    final RegisterLnurlPayResponse registrationResponse = await lnAddressService.register(
+    _logger.info('Preparing register LNURL Webhook request.');
+    final bool isUpdating = baseUsername != null;
+    final bool isLnUrlWebhookRegistered = await breezPreferences.isLnUrlWebhookRegistered;
+    final bool isNewRegistration = !(isUpdating && isLnUrlWebhookRegistered);
+
+    final String? username = baseUsername ?? await _resolveUsername(isNewRegistration: isNewRegistration);
+
+    // Register without retries if this is an update to existing LNURL Webhook
+    if (!isNewRegistration) {
+      return await _attemptRegisterLnurlWebhook(
+        pubKey: pubKey,
+        webhookUrl: webhookUrl,
+        username: username,
+      );
+    }
+
+    // Register with retries if LNURL Webhook hasn't been registered yet
+    return await _registerWithRetries(
       pubKey: pubKey,
-      request: request,
+      webhookUrl: webhookUrl,
+      username: username!,
     );
+  }
 
-    final String? username = request.username;
-    if (username != null && username.isNotEmpty) {
-      await breezPreferences.setLnAddressUsername(username);
+  // TODO(erdemyerebasmaz): Optimize if current retry logic is insufficient
+  // If initial registration fails, up to [_maxRetries] registration attempts will be made on opening [ReceiveLightningAddressPage].
+  // If these attempts also fail, the user can retry manually via a button, which will trigger another registration attempt with [_maxRetries] retries.
+  //
+  // Future improvements could include:
+  // - Retrying indefinitely with intervals until registration succeeds
+  // - Explicit handling of [UsernameConflictException] and LNURL server connectivity issues
+  // - Randomizing the default profile name itself after a set number of failures
+  // - Adding additional digits to the discriminator
+  Future<RegisterRecoverLnurlPayResponse> _registerWithRetries({
+    required String pubKey,
+    required String webhookUrl,
+    required String username,
+  }) async {
+    final String baseUsername = username;
+    for (int retryCount = 1; retryCount <= _maxRetries; retryCount++) {
+      try {
+        _logger.info('Attempt $retryCount/$_maxRetries with username: $username');
+        return await _attemptRegisterLnurlWebhook(
+          pubKey: pubKey,
+          webhookUrl: webhookUrl,
+          username: username,
+        );
+      } on UsernameConflictException {
+        _logger.warning('Username conflict for: $username.');
+        username = UsernameGenerator.generateUsername(baseUsername, retryCount);
+      } catch (e) {
```

Comment:
From what I understand either the `on UsernameConflictException` or `catch` branch will execute. If we're unlucky enough to get a conflict 3 times, I don't see it throwing an error. Did I miss something?

Reply:
I think this is resolved with: https://github.com/breez/misty-breez/commit/9d4bf585376a707b8ff82e553b30caf1022f871e